core/domain/suggestion_domain.py

```
@@ -72,3 +77,73 @@ def to_dict(self):
             'change_cmd': self.change_cmd,
             'score_category': self.score_category
         }
+
+    def validate(self):
+        """Validates the suggestion object. Each subclass must implement
+        this function
+        """
+        pass
+
+
+class SuggestionEditStateContent(BaseSuggestion):
+    """Domain object for a suggestion of type SUGGESTION_EDIT_STATE_CONTENT.
+
+    Attributes:
+        suggestion_id: str. The ID of the suggestion.
+        suggestion_type: str. The type of the suggestion.
+        target_type: str. The type of target entity being edited.
+        target_id: str. The ID of the target entity being edited.
+        target_version_at_submission: int. The version number of the target
+            entity at the time of creation of the suggestion.
+        status: str. The status of the suggestion.
+        author_id: str. The ID of the user who submitted the suggestion.
+        assigned_reviewer_id: str. The ID of the user assigned to
+            review the suggestion.
+        final_reviewer_id: str. The ID of the reviewer who has accepted/rejected
+            the suggestion.
+        change_cmd: dict. The actual content of the suggestion.
+        score_category: str. The scoring category for the suggestion.
+    """
+
+    def __init__(
+            self, suggestion_id, target_id, target_version_at_submission,
+            status, author_id, assigned_reviewer_id, final_reviewer_id,
+            change_cmd, score_category):
+        """Initializes a Suggestion object of type
+        SUGGESTION_EDIT_STATE_CONTENT.
+        """
+        super(SuggestionEditStateContent, self).__init__(
+            suggestion_id, suggestion_models.SUGGESTION_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION, target_id,
+            target_version_at_submission, status, author_id,
+            assigned_reviewer_id, final_reviewer_id, change_cmd,
+            score_category)
+
+    def validate(self):
```

Other validation methods (see e.g. exp_domain) raise errors in the case of a validation failure; I think we should be consistent especially if the invalidity is not something we would expect to happen in the normal course of events (e.g. it should already be handled by the frontend).

---

Changed it to raise errors instead of returning bool. Also I added a comprehensive set of checks for the object.